lib/simplecov/defaults.rb

```diff
@@ -4,7 +4,7 @@
 SimpleCov.profiles.define "root_filter" do
   # Exclude all files outside of simplecov root
   add_filter do |src|
-    !(src.filename =~ /^#{Regexp.escape(SimpleCov.root)}/i)
+    !src.filename.downcase.start_with?(SimpleCov.root.downcase)
```

Comment:
I'm trying to think if there's any chance src.filename might be nil

Also, can you try benchmarking with `/^#{Regexp.escape(SimpleCov.root)}/io` (the o means it will only compile the interpolated value once), and also maybe `\A` instead of `^`, and also maybe `ROOT_FILTER = /^{Regexp.escape(SimpleCov.root)}/i)`?